R/regression.R

```diff
@@ -218,6 +219,30 @@ Regression <- function(formula,
         stop("'weights' and 'data' are required to have the same number of observations. They do not.")
     if (!is.null(subset) & length(subset) > 1 & length(subset) != nrow(data))
         stop("'subset' and 'data' are required to have the same number of observations. They do not.")
+    missing.variables <- apply(data, 2, function(x) all(is.na(x)))
+    if(any(missing.variables))
+    {
+        missing.variable.names <- colnames(data)[missing.variables]
+        if (outcome.name %in% missing.variable.names)
+            stop("Response variable is entirely missing (all observed values of the variable are missing).")
+        else {
+            # Update data and formula
+            formula.terms <- terms.formula(input.formula, data = data)
+            terms.to.drop <- unique(unlist(sapply(missing.variable.names, grep, x = attr(formula.terms, "term.labels"))))
+            input.formula <- update(input.formula, drop.terms(formula.terms, terms.to.drop, keep.response = TRUE))
+            interaction.formula.terms <- terms.formula(formula.with.interaction, data = data)
+            interaction.terms.to.drop <- unique(unlist(sapply(missing.variable.names, grep,
+                                                              x = attr(interaction.formula.terms, "term.labels"))))
+            formula.with.interaction <- update(formula.with.interaction, drop.terms(interaction.formula.terms,
+                                                                                    interaction.terms.to.drop,
+                                                                                    keep.response = TRUE))
+            formula <- input.formula
+            data <- data[, !missing.variables]
+            missing.variable.names <- paste0(missing.variable.names, collapse = ", ")
+            warning("Data has variable(s) that are entirely missing values (all observed values of the variable are missing). ",
+                    "These variable(s) have been removed from the analysis (", missing.variable.names, ").")
```

Comment:
Should the variable names be followed by a colon instead of wrapped in parentheses? i.e.:
These variable(s) have been removed from the analysis: variable.name1, variable.name2, variable.name3.

Reply:
Made suggested changes in https://github.com/Displayr/flipRegression/pull/5